components/membership/registration-impl/src/main/kotlin/net/corda/membership/impl/registration/dynamic/member/DynamicMemberRegistrationService.kt

```
@@ -452,7 +453,7 @@ class DynamicMemberRegistrationService @Activate constructor(
             tenantId: String,
             expectedCategory: String,
         ): List<CryptoSigningKey> =
-            cryptoOpsClient.lookup(tenantId, keyIds).also { keys ->
+            cryptoOpsClient.lookupKeysByIds(tenantId, keyIds.map { ShortHash.of(it) }).also { keys ->
```

If the short hash is invalid we need to throw the correct exception

---

I went back to see what is calling this, so it is not a rest resource. Currently that will throw if hex string is not valid (Please see `ShortHash.of`).

Unless you mean we need to make it more strict and use `ShortHash.parse`?

---

If the register throws a random exception, we will retry it several times. If the key is invalid, we should not retry it at all. To stop the retry loop, if the key ID is not a short hash, we should throw an `InvalidMembershipRegistrationException`.